scripts/linters/pylint_extensions.py

```diff
@@ -1959,6 +2018,244 @@ def process_tokens(self, tokens):
+        outside_args_section = True
+        any_present_inside_arg_section = False
+        any_present_inside_return_section = False
+        any_present_in_function_def = False
+        multi_line_import = False
+        single_line_import = False
+        import_line_num = 0
+        args_section_end_line_num = 0
+
+        any_type_regex = r'^# Here we used type Any because'
+
+        for (token_type, token, (line_num, _), _, line) in tokens:
+            line = line.strip()
+
+            if token_type == tokenize.COMMENT:
+                if re.search(any_type_regex, line):
+                    any_type_comment_present = True
+
+            # Checking if single-line import is present.
+            # Eg: from typing import Any.
+            if token_type == tokenize.NAME:
+                if token == 'import':
+                    single_line_import = True
+                    import_line_num = line_num
+
+            # Checking if multi-line import is present.
+            # Eg: from typing import (
+            #   Any, Callable, Dict, FrozenSet, Iterator, List, Set,
+            #   Tuple, Type, cast
+            # )
+            if token_type == tokenize.OP:
+                if single_line_import and token == '(':
+                    multi_line_import = True
+                    single_line_import = False
+                if multi_line_import and token == ')':
+                    multi_line_import = False
+
+            # Checking if linters are in argument-section, return-section or
+            # outside of the function definition.
+            # Eg:
+            #       <outside function definition>
+            #   def func(<argument-section>) -> <return-section>:
+            #       <outside function definition>.
+            if token_type == tokenize.NAME:
+                if token == 'def':
+                    outside_function_def = False
+                    func_def_start_line = line_num
+                    outside_args_section = False
+
+            if token_type == tokenize.OP:
+                if token == '->':
+                    outside_args_section = True
+                    args_section_end_line_num = line_num
+                if outside_args_section and token == ':':
+                    outside_function_def = True
+
+            # Checking if Any type is present in function definition or not.
+            if token_type == tokenize.NAME and token == 'Any':
+                if not outside_args_section:
+                    any_present_inside_arg_section = True
+                elif (
+                    outside_args_section and
+                    args_section_end_line_num == line_num
+                ):
+                    any_present_inside_return_section = True
+
+            if (
+                any_present_inside_arg_section or
+                any_present_inside_return_section
+            ):
+                any_present_in_function_def = True
+
+            if outside_function_def:
```

Comment:
Isn't `outside_function_def` basically `!any_present_in_function_def`? If not then I think I need more explanation and better variable names.

Reply:
Also should we have similar check for `object` (like whether it is inside function definition or not)? I would imagine the `Any` and `object` check being basically the same except for the import part.

Reply:
> Also should we have similar check for object

Done!

> Isn't outside_function_def basically !any_present_in_function_def 


No, `any_present_in_function_def` signifies that any type was present in function signature or not, whereas `outside_function_def` signifies that linters are in function signature scope or not -- if linters are in function signature block then it is False otherwise True.

because we are throwing errors only when we are outside of the function signature. By this I mean, if we encountered an exceptional type inside function signature then there we are just keeping a record that yeah we have an exceptional type inside function signature and when we come outside of the function signature then we throw an error.

suppose 
```python 

some_any_dict: Dict[str, Any] = {} # Here we throw error directly because we are outside of function signature. 

# some other content of the modules.

def foo(arg: Any) - > None:      # So, (Here we are not throwing an error, because we are in function signature but we are keeping a record that we have exceptional type inside signatures so as soon as we come out of the signature we throw an error ) Here we encountered expectational type inside args-section (explained in code comment ) so, any_present_inside_arg_section is True and arg-section belongs to function signature so any_present_in_function_signature is also true.
   pass   # Here we throw an error that any was encountered inside function signature.

def foo1(arg1: str) -> str:
    var: Any = None   # Here we are throwing an error because we are in function definition not function signature.
    return 'hi'



```

`if outside_function_def` basically means that we are not in the function signature. Because for function signatures we are checking separately as mentioned above. 

Btw outside_function_def changed to outside_function_signature_block and `any_present_in_function_def` is changed to  ` any_present_in_function_signature`.